src/app/auth/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { inject } from '@angular/core';
+import { CanActivateFn, Router } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+import { take } from 'rxjs/operators';
+
+export const authGuard: CanActivateFn = (route, state) => {
+  const authService = inject(AuthService);
+  const router = inject(Router);
+
+  authService.user$.pipe(take(1)).subscribe((user) => {
+    if (user) {
+      authService.currentUserSig.set({
+        //! is added because it is pretty much guaranteed that the user is logged in at this point
```

Comment:
I would suggest using a ternary like` user.email ? user.email : ""`. Incase the user is logged in but the api returns null values for the user.